server/interfaces/interfaces.go

```diff
@@ -754,13 +753,20 @@ type WebhookData struct {
 	// Ex: "main"
 	TargetRepoDefaultBranch string
 
-	// TargetBranch is the branch associated with the event that determines whether
-	// actions should be triggered. For push events this is the branch that was
-	// pushed to. For pull_request events this is the base branch into which the PR
-	// branch is being merged.
+	// TargetBranch is the base branch into which the PR branch is being merged.
+	// For non-pull request events, this can be empty, or can equal `PushedBranch`.
 	// Ex: "main"
 	TargetBranch string
 
+	// MergeCommitSHA is the SHA generated by merging the pushed branch into the
+	// target branch. This behavior is used to guarantee that CI runs on up-to-date
+	// refs. It will be empty for non-pull-request events.The merge commit SHA is
+	// generated by a background job, so can be empty for pull-request events if
+	// the background job has not completed, or if the webhook provider does not
+	// provide a generated merge commit. In this case, it should be manually
+	// generated by merging the pushed and target branches.
+	MergeCommitSHA string
```

Comment:
any idea how reliable this merge_commit_sha is? if we're unsure, can we put it behind a flag and test in dev for a while?

Reply:
err I guess if it's non-empty, then it should exist remotely, otherwise they wouldn't know the SHA?

Reply:
If it exists on the webhook data, we should be able to reliably fetch it. I'll merge this now, so we'll have until Thursday to see how this does in dev